Integrations/GoogleKeyManagementService/GoogleKeyManagementService.py

```diff
@@ -0,0 +1,854 @@
+            'GenerateTime': datetime.fromtimestamp(int(res.primary.generate_time.seconds)).strftime(DEMISTO_DATETIME_FORMAT)
+        }
+
+    return key_context
+
+
+def crypto_key_to_json(crypto_key: Any) -> Dict:
+    """Creates a json dict from `~google.cloud.kms_v1.types.CryptoKey` instance to use as raw response.
+
+    Args:
+        crypto_key(Any): `~google.cloud.kms_v1.types.CryptoKey` instance.
+
+    Returns:
+        A json Dict containing the raw response.
+    """
+    key_json = {
+        'name': crypto_key.name,
+        'purpose': enums.CryptoKey.CryptoKeyPurpose(crypto_key.purpose).name,
+        'create_time': {
+            'seconds': crypto_key.create_time.seconds,
+            'nanos': crypto_key.create_time.nanos
+
+        },
+        'next_rotation_time': {
+            'seconds': crypto_key.next_rotation_time.seconds,
+            'nanos': crypto_key.next_rotation_time.nanos
+        },
+        'rotation_period': {
+            'seconds': crypto_key.rotation_period.seconds
+        },
+        'labels': arg_dict_creator(str(crypto_key.labels)[1:-1]),
+        'version_template': {
+            'protection_level': enums.ProtectionLevel(crypto_key.version_template.protection_level).name,
+            'algorithm': enums.CryptoKeyVersion.CryptoKeyVersionAlgorithm(crypto_key.version_template.algorithm).name,
+        }
+    }
+
+    if crypto_key.primary:
+        key_json['primary'] = {
+            'name': crypto_key.primary.name,
+            'state': enums.CryptoKeyVersion.CryptoKeyVersionState(crypto_key.primary.state).name,
+            'create_time': {
+                'seconds': crypto_key.primary.create_time.seconds,
+                'nanos': crypto_key.primary.create_time.nanos
+            },
+            'protection_level': enums.ProtectionLevel(crypto_key.primary.protection_level).name,
+            'algorithm': enums.CryptoKeyVersion.CryptoKeyVersionAlgorithm(crypto_key.primary.algorithm).name,
+            'generate_time': {
+                'seconds': crypto_key.primary.generate_time.seconds,
+                'nanos': crypto_key.primary.generate_time.nanos
+            }
+        }
+
+    return key_json
+
+
+def demisto_args_extract(client: Client, args: Dict[str, Any]) -> Tuple[str, str, str, str]:
+    """Extracts IDs to use for KMS functions.
+
+    Args:
+        args(dict): Demisto arguments.
+        client(Client): User Client.
+
+    Returns:
+        A tuple containing strings representing the required IDs.
+    """
+    project_id = client.project
+
+    location_id = args.get('location_id')
+    if location_id == 'default':
+        location_id = client.location
+
+    key_ring_id = args.get('key_ring_id')
+    if key_ring_id == 'default':
+        key_ring_id = client.key_ring
+
+    crypto_key_id = args.get('crypto_key_id')
+    return str(project_id), str(location_id), str(key_ring_id), str(crypto_key_id)
+
+
+def get_update_mask(args: Dict[str, Any]) -> Dict:
```

Comment:
still not sure why you need this function... let's discuss in person

Reply:
This is part of the API's update request - look here for examples: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/kms/api-client/snippets.py